### PR TITLE
Ensure #ordinalize to work with all the numbers from 0 to 100

### PR DESCRIPTION
**Pull request by jodosha**
@sondnm reported in [chat](https://gitter.im/dry-rb/chat?at=5a0eec63e606d60e34d6e059): 

```ruby
inflector = Dry::Inflector.new
inflector.ordinalize(17) # nil
inflector.ordinalize(24) # nil
inflector.ordinalize(25) # nil
inflector.ordinalize(34) # nil
```

This PR fixes the problem.

### CODE REVIEW

File: lib/dry/inflector.rb [hunk 1]
```diff
@@ -273,7 +273,7 @@ def uncountable?(input)
 
     # @since 0.1.0
     # @api private
-    ORDINALIZE_TH = (4..16).to_set.freeze
+    ORDINALIZE_TH = (4..16).each_with_object({}) { |n, ret| ret[n] = true }.freeze
```
Comment (Ptico) [inline lib/dry/inflector.rb]: Basically `Set` is the same as having key: true hash. And we are using it in the `Inflections` for performance reasons. So, why removing it?

Comment (abinoam) [reply lib/dry/inflector.rb]: Agree @Ptico.
The only fix needed is the
```ruby
else "#{number}th"
```

By the way... this bug affects Inflecto/Flexus!!! 😱

**Comment by abinoam** [comment]
By the way @jodosha, the current Rails implementation has a clearer semantic.

We put 4 to 16 as exceptions.
When the only real exceptions are 11 to 13.
4 to 10 and 14 to 16 falls into the "not ending with 1, 2 or 3" category, the else branch.

Well, I don't know If I'm making myself clear.
What I mean is... Looking at the ActiveSupport code is clearer to me that all numbers ending in 1, 2, or 3 has a special end aside from 11, 12 and 13.

We could take this opportunity to improve this. What do you think?

```
    # Returns the suffix that should be added to a number to denote the position
    # in an ordered sequence such as 1st, 2nd, 3rd, 4th.
    #
    #   ordinal(1)     # => "st"
    #   ordinal(2)     # => "nd"
    #   ordinal(1002)  # => "nd"
    #   ordinal(1003)  # => "rd"
    #   ordinal(-11)   # => "th"
    #   ordinal(-1021) # => "st"
    def ordinal(number)
      abs_number = number.to_i.abs

      if (11..13).include?(abs_number % 100)
        "th"
      else
        case abs_number % 10
        when 1; "st"
        when 2; "nd"
        when 3; "rd"
          else    "th"
        end
      end
    end

    # Turns a number into an ordinal string used to denote the position in an
    # ordered sequence such as 1st, 2nd, 3rd, 4th.
    #
    #   ordinalize(1)     # => "1st"
    #   ordinalize(2)     # => "2nd"
    #   ordinalize(1002)  # => "1002nd"
    #   ordinalize(1003)  # => "1003rd"
    #   ordinalize(-11)   # => "-11th"
    #   ordinalize(-1021) # => "-1021st"
    def ordinalize(number)
      "#{number}#{ordinal(number)}"
    end
```

**Comment by abinoam** [comment]
And...

I would not indent the `else` beyond the `when`s. Ugly!

The foolowing is prettier.
```ruby
        case abs_number % 10
        when 1; "st"
        when 2; "nd"
        when 3; "rd"
        else    "th"
        end
```

**Comment by Ptico** [comment]
@abinoam Looks like (4..16) was introduced by me, i dunno why 🤦‍♂️

**Comment by jodosha** [comment]
@Ptico @abinoam Thanks for the really good feedback. I wasn't sure about the `ORDINALIZE_TH` values, but with `11..13` the tests are passing. Please feel free to cleanup, according your experience. 👍 

---

Regarding the `Set` vs `Hash` it's a perf trick: `Set` wraps a `Hash` to guarantee unique results. So we can avoid the wrapping at all and use `Hash#key?`, which is a bit faster than `Set#include?`. See https://gist.github.com/jodosha/743d5c8672ec8d90358f7cbc50f39806

**Comment by abinoam** [comment]
> which is a bit faster than Set#include?

👍 So let's stay with Hash

I'll take the opportunity to do the 11..13 trick.

**Comment by jodosha** [comment]
Thank you all. Great team work. High 5! ✋

**Comment by abinoam** [comment]
High 5 ✋

**Pull request merged**